### ISSUE TITLE
Material Instances: Visual discrepancies between VRM0 and VRM1 versions of the same avatar

### ISSUE THREAD

**Issue by andreiagmu** [ruyo/VRM4U#438]
Unreal Engine v5.4.4
VRM4U v20241007

**Describe the bug**
After importing a `VRM1` avatar, there are quite a few inconsistencies in the Material Instances, compared to when the same avatar is imported as `VRM0`.

Note: The avatar from my tests is exported from `VRoid Studio` (v1.29.55 - "Dress-up Feature" open beta), in both VRM0 and VRM1 format.
Also, in both VRM0 and VRM1 imports to UE, I'm using the **default** VRM4U import options (e.g.: `Force Opaque = false`, `Force TwoSided = false`).

I noticed some values in the `10_Main` parameter group differ between VRM0 and VRM1 (e.g.: `mtoon_ShadeShift`, `mtoon_ShadeToony`).
I'm not sure if those are the intended parameter values for VRM1, or if the Material Instances should've been imported with the exact same values as VRM0.
Anyway, in my case, this isn't the main issue (apparently). Changing those parameters to their VRM0 equivalents doesn't seem to change the visuals much.

The main issue I noticed is that many (most? all?) Material Instances that were originally `TwoSided` in VRM0, aren't `TwoSided` anymore in VRM1. See the Material Instances' `Parent`, at the bottom of my screenshots.
Furthermore, some Material Instances that were `Translucent` _and_ `TwoSided` in VRM0, are now simply `Opaque` in VRM1.
The visual discrepancies are noticeable in the clothes (inner side) and, especially, in the Glasses accessory from VRoid Studio.

I'm aware I can modify the VRM1 Material Instances after importing the avatar, to achieve the same result as VRM0.
But I'd like to avoid any kind of extra setup, that I'd potentially have to repeat with _many_ imported avatars in my project.
Especially as the VRM0 Material Instances are already correct and "just work" out of the box.

Note: I'm using VRM1 in my project, to make my avatars more "future-proof" (in theory), and to be able to use the new "Dress-up Feature" from VRoid Studio, that's only available for VRM1 avatars.

**Comment by andreiagmu** [comment]
In the following screenshot pairs, the first image shows the VRM0 avatar, and the second image shows the VRM1 avatar.

---
![capture_20241106_021402_055_edit](https://github.com/user-attachments/assets/190552ec-37f7-44a8-a123-f2a281d121c4)

![capture_20241106_021526_056_edit](https://github.com/user-attachments/assets/06612db9-8a76-4056-91bf-ee9f2869dc89)

---

**Comment by andreiagmu** [comment]
I guess the changed values I noticed in the `10_Main` parameter group are intentional, as the VRM1 avatar has a stronger "cel/toon shading effect" by default, when compared to the VRM0 avatar.
It would be nice to add an (optional) VRM1 import setting to use the old/legacy (?) VRM0 values for the Material Instances.

My issues with the (post-import default) VRM1 Material Instances are mainly:
- The loss of certain light/translucency effects (e.g.: Glasses).
- Certain Material Instances that make sense to be TwoSided by default (e.g.: Clothes) aren't TwoSided anymore.

**Comment by ruyo** [comment]
Hi.
Thanks for the detailed report. Some glitches will be fixed next time.

However, VRM0 and VRM1 output by VRoidStudio look different to begin with. So it is not valid to compare them.
If you want to compare their appearance accurately, you need to compare them with the state read by UniVRM.

**Comment by andreiagmu** [comment]
I see, thanks for the info!
As you suggested, I compared my avatar's VRM0 and VRM1 versions, using UniVRM (in Unity).

To clarify, I don't mind the MToon shading differences between the two VRM outputs. I generally prefer the VRM1 version (in both Unreal and Unity).
So, I guess the "changed Material Instance parameters" in Unreal are correct, following the VRM1 output from VRoid Studio. 👍 

I did notice when using UniVRM, the materials that were double-sided and/or translucent in VRM0 _kept being_ double-sided and/or translucent in VRM1.
So, with UniVRM, the glasses still have the light/translucency effects in VRM1, clothes are correctly double-sided in VRM1, etc.
